conda/gateways/connection/adapters/jlap/repodata_proxy.py

```diff
@@ -114,65 +83,58 @@ def apply_patches(data, patches, have, want):
     return data
 
 
-@contextlib.contextmanager
-def timeme(message=""):
-    begin = time.time()
-    yield
-    end = time.time()
-    log.debug(f"{message}{end-begin:0.02f}s")
+def fetch_repodata_json(cache_path: Path, repodata_url: str, ses=None) -> Tuple[Path, str]:
+    """
+    Fetch new repodata.json; cache to a gzip'd file
+    """
+    if ses is None:
+        ses = requests
 
+    with tempfile.NamedTemporaryFile(dir=consts.CACHE_DIR, delete=False) as outfile:
+        with gzip.open(outfile, "w") as compressed:
+            response = ses.get(repodata_url, stream=True)
+            response.raise_for_status()
+            hash_val = hash_func()
 
-def fetch_repodata_json(cache_path, upstream, session=requests):
-    """
-    Fetch new repodata.json; cache to a gzip'd file.
+            for chunk in response.iter_content(consts.JLAP_CHUNK_SIZE):
+                hash_val.update(chunk)
+                compressed.write(chunk)
 
-    Return (path, digest)
-    """
+            cache_path.parent.mkdir(parents=True, exist_ok=True)
 
-    with tempfile.NamedTemporaryFile(dir=CACHE_DIR, delete=False) as outfile:
-        compressed = gzip.open(outfile, "w")
-        response = session.get(upstream, stream=True)
-        response.raise_for_status()
-        hash = hash_func()
-        for chunk in response.iter_content(CHUNK_SIZE):
-            hash.update(chunk)
-            compressed.write(chunk)
-
-        cache_path.parent.mkdir(parents=True, exist_ok=True)
-        compressed.close()
+        # tmpdir -> cache_dir
         os.replace(outfile.name, cache_path)
 
-    return cache_path, hash.digest()
+    return cache_path, hash_val.digest()
 
 
-class DigestReader:
+class DigestReader(NamedTuple):
     """
     Read and hash at the same time.
     """
 
-    def __init__(self, fp):
-        self.fp = fp
-        self.hash = hash_func()
+    fp: gzip.GzipFile
+    hash: hashlib.blake2b
 
-    def read(self, bytes=None):
-        buf = self.fp.read(bytes)
+    def read(self, bytes_val=None):
+        buf = self.fp.read(bytes_val)
         self.hash.update(buf)
+
         return buf
 
 
 def patch_files(cache_path: Path, jlap_path: Path):
     """
     Return patched version of cache_path, as a dict
     """
-    jlap_lines = []
     with jlap_path.open("rb") as fp:
         jlap = truncateable.JlapReader(fp)
-        jlap_lines = list(obj for obj, _ in jlap.readobjs())
+        jlap_lines = tuple(obj for obj, _ in jlap.read_objs())
         assert "latest" in jlap_lines[-1]
 
     meta = jlap_lines[-1]
     patches = jlap_lines[:-1]
-    digest_reader = DigestReader(gzip.open(cache_path))
+    digest_reader = DigestReader(gzip.open(cache_path), hash_func())
```

Comment:
the repodata.json of course compresses extremely well. fast compression can save disk time. it would be fun to time that here, and when we write